core/src/main/java/org/infinispan/protostream/impl/TagWriterImpl.java

```diff
@@ -1153,9 +1153,7 @@ public OutputStreamRandomAccessEncoder(RandomAccessOutputStream out) {
       void writeUInt32Field(int fieldNumber, int value) throws IOException {
          int pos = out.getPosition();
          int tag = WireType.makeTag(fieldNumber, WireType.WIRETYPE_VARINT);
-         int tagLen = varIntBytes(tag);
-         int valueLen = varIntBytes(value);
-         out.ensureCapacity(pos + tagLen + valueLen);
+         out.ensureCapacity(pos + 5 + 5);
```

Comment:
See other comment. I don't think we can assume 5 bytes for the tag as it can be negative, I think it should be fine for the value though.

Reply:
Same here the tag is only an int so its max size is 5.